src/contracts/StakingChef.sol

```diff
@@ -95,73 +77,72 @@ contract StakingChef is Chef {
         setLastRewardBlock(block.number);
     }
 
-// --------------------------------------------------------------------- Manage rewards and Principal
+    // --------------------------------------------------------------------- Manage rewards and Principal
 
-    function claim(uint256 stakeIndex) public {
-        Staker storage stake = _stakes[stakeIndex];
-        harvest(stakeIndex);
+    function claim() public {
+        Staker storage stake = staker[msg.sender];
+        harvest();
 
         require(stake.claimable != 0, "No claimable rewards to withdraw");
-        require(stake.stakeAddress == _msgSender(), "Sender can not claim");
 
         SafeERC20.safeTransfer(IERC20(getMetricToken()), msg.sender, stake.claimable);
         stake.claimable = 0;
 
-        emit Withdraw(msg.sender, stakeIndex, stake.claimable);
+        emit TransferPrincipal(msg.sender, stake, stake.claimable);
     }
 
-    function withdrawPrincipal(uint256 stakeIndex) public {
-        Staker storage stake = _stakes[stakeIndex];
-
+    function withdrawPrincipal() public {
+        Staker storage stake = staker[msg.sender];
         require(stake.metricAmount != 0, "No Metric to withdraw");
-        require(stake.stakeAddress == _msgSender(), "Sender can not withdraw");
 
         SafeERC20.safeTransfer(IERC20(getMetricToken()), msg.sender, stake.metricAmount);
         stake.metricAmount = 0;
```

Comment:
since this is storage I'm guessing you can use stake as a setter but if it was memory you'd have to use staker[msg.sender].metricAmount = 0; 
?